File: src/main/java/com/polykhel/billnext/service/mapper/CategoryMapper.java
package com.polykhel.billnext.service.mapper;

import com.polykhel.billnext.domain.Category;
import com.polykhel.billnext.service.dto.CategoryDTO;
import org.mapstruct.BeanMapping;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.Named;

/**
 * Mapper for the entity {@link Category} and its DTO {@link CategoryDTO}.
 */
@Mapper(componentModel = "spring", uses = { UserMapper.class })
public interface CategoryMapper extends EntityMapper<CategoryDTO, Category> {
    @Mapping(target = "user", source = "user", qualifiedByName = "id")
    CategoryDTO toDto(Category s);

    @Named("id")
    @BeanMapping(ignoreByDefault = true)
    @Mapping(target = "id", source = "id")
    CategoryDTO toDtoId(Category category);
}

File: src/main/kotlin/com/polykhel/billnext/service/mapper/CategoryMapper.kt
package com.polykhel.billnext.service.mapper

import com.polykhel.billnext.domain.Category
import com.polykhel.billnext.service.dto.CategoryDTO
import org.mapstruct.*

/**
 * Mapper for the entity [Category] and its DTO [CategoryDTO].
 */
@Mapper(componentModel = "spring", uses = [UserMapper::class])
interface CategoryMapper :
    EntityMapper<CategoryDTO, Category> {

    @Mappings(
        Mapping(target = "user", source = "user", qualifiedByName = ["id"])
    )
    override fun toDto(s: Category): CategoryDTO

    @Named("id")
    @BeanMapping(ignoreByDefault = true)
    @Mappings(
        Mapping(target = "id", source = "id")
    )
    fun toDtoId(category: Category): CategoryDTO
}
